lib/chef/resource/aws_cache_cluster.rb

```diff
@@ -0,0 +1,31 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+class Chef::Resource::AwsCacheCluster < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type AWS::ElastiCache
+
+  attribute :cluster_name, kind_of: String, name_attribute: true
+  attribute :az_mode, kind_of: String
```

Comment:
I know its more verbose, but I think the readability is worth it for `:availability_zone_mode`